### PR TITLE
fix for issue 536: Jackson JsonAnySetter support to for dynamic fields using crnk-client

### PR DESCRIPTION
**Pull request by magg**
Please see issue https://github.com/crnk-project/crnk-framework/issues/536

### CODE REVIEW

File: crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java [hunk 1]
```diff
@@ -123,7 +126,18 @@ protected void setAttributes(Resource resource, Object entity, ResourceInformati
 					setAttribute(resource, field, value);
 				}
 			}
+
+		}
+
+		if (resourceInformation.getAnyFieldAccessor() != null) {
+			AnyResourceFieldAccessor anyFieldAccessor = resourceInformation.getAnyFieldAccessor();
+			Object o = anyFieldAccessor.getValues(entity);
```
Comment (remmeier) [inline crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java]: The anyFieldAccessor could return a Map directly I guess?

**Comment by remmeier** [comment]
looks great, thx!  added one small comment. Potentially a one small test case on the client side would be good to make sure it remains working. For example in BasicRepositoryAccessTestBase that is inherited from RepositoryAccessClientTest and adding the any getter/setter to one of the types like Schedule..

File: crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java (hunk 1 above)
Comment (magg) [reply crnk-core/src/main/java/io/crnk/core/engine/internal/document/mapper/ResourceMapper.java]: OK I will update it, sound good

**Comment by magg** [comment]
Hi @remmeier I changed the code according to your comment and added the test, but it's failing, could you please take a look?

**Comment by magg** [comment]
so there's a weird bug... when run this test it fails
```
    @Test
    public void testCreateAnyFields() {
        Schedule schedule = new Schedule();
        schedule.setName("mySchedule");
        schedule.setAnyFields("randomField1", 1234);
        schedule.setAnyFields("randomField2", "test");

        scheduleRepo.create(schedule);

        QuerySpec querySpec = new QuerySpec(Schedule.class);
        ResourceList<Schedule> list = scheduleRepo.findAll(querySpec);
        Assert.assertEquals(1, list.size());
        schedule = list.get(0);
        Assert.assertNotNull(schedule.getId());
        Assert.assertEquals(1234, schedule.getAnyFields().get("randomField1"));
        Assert.assertEquals("test", schedule.getAnyFields().get("randomField2"));

    }

```


scheduleRepo.findAll(querySpec) does not bring the extra fields, but scheduleRepo.create(schedule) does return them. I committed the code where i check scheduleRepo.create(schedule) output

Do you have any ideas if there's a difference in the handling of create and findAll? ResourceUpsert.setAttribute does the handling in the back-end right?

**Comment by magg** [comment]
So @remmeier I think the problem with the findOne/findAll methods rely on DefaultQueryPathResolver class, it has a variable allowUnknownAttributes = false;

I found a workaround to fix the findOne/findAll methods, 

in my code this fixed the issue by setting a new DefaultQuerySpecUrlMapper in the CrnkClient

```       
DefaultQuerySpecUrlMapper urlMapper = new DefaultQuerySpecUrlMapper();
urlMapper.setAllowUnknownAttributes(true);
client.setUrlMapper(urlMapper);
```

But I think this cannot be added in BasicRepositoryAccessTestBase without it breaking many tests.

**Comment by remmeier** [comment]
ResourceUpset does the update processing logic, whereas ResourceMapper translates objects to json structures. So they work in opposite directions but maybe could be merged at some point for clarity.

allowUnknownAttributes is only involved for urls (sorting, filtering). Ideally it would back-off automatically is the resourceInformation has any setters. 

I can take a look this evening if it does not clear up.

**Comment by magg** [comment]
added a simple check in DefaultQueryPathResolver.java to check jsonanysetter too

`if (allowUnknownAttributes || anyFieldAccessor != null) `

**Comment by remmeier** [comment]
thx a lot!

**Pull request merged**